worker/src/RTC/Router.cpp

```diff
@@ -710,7 +710,7 @@ namespace RTC
 				if (!mid.empty())
 					packet->UpdateMid(mid);
 
-				consumer->SendRtpPacket(packet, &clonedPacket);
+				consumer->SendRtpPacket(packet, clonedPacket);
```

Comment:
I wonder why don't why do:

````
Consumer::SendRtpPacket(RTC::RtpPacket::SharedPtr& packet);
````

So we don't even need to clone the original packet but just store a reference (within the shared_ptr). In the end we are modifying the SSRC, Cseq of it before sending.

Reply:
This smart pointer is `nullptr`, see line 703 above. The idea was to store cloned packet in `clonedPacket` for subsequent calls, while avoiding cloning at all if it is not needed (audio packets or RTX disabled). The way you changed it here would work if we call `.reset()` later (earlier revision wasn't using smart pointers, this is why it was done this way).

Reply:
It's working as expected within my latest changes.

* We create `clonedPacket`, a `SharedPtr` with `nullptr` content.
* We pass it to consumers.
* The first consumer that needs to store it, will actually clone the original packet, and add set the `RtpPacket` pointer into `clonedPacket`. At which stage `clonedPacket` will have two uses.
* When we get out of the scope if this method in `Router`, the reference counter will be decremented and upon the last decrement it will be released, no need to call `.reset()`